### PR TITLE
Event log 2

### PR DESCRIPTION
**Pull request by RONNCC**
#146

### CODE REVIEW

**Comment by purelogiq** [comment]
I'm going to review this later tomorrow

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  def approve
+    @event = Event.find(params[:id])
+    @event.is_done = !@event.is_done
+    @event.save
+    respond_with @event, location: -> {events_path}
+  end
+
+  # GET /events/1/edit
+  def edit
+    @event = Event.find(params[:id])
```
Comment (minigeek) [inline app/controllers/events_controller.rb]: Cancan should do this automatically, right? (find the event)

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# **`description`**    | `text(65535)`      |
+# **`event_type_id`**  | `integer`          |
+# **`id`**             | `integer`          | `not null, primary key`
+# **`is_done`**        | `boolean`          |
+# **`updated_at`**     | `datetime`         |
+#
+
+class Event < ActiveRecord::Base
+      belongs_to :event_type
+      scope :sidebar_viewable, -> { joins(:event_type).where(is_done:false).where('event_types.display = ?', true) }
```
Comment (pkoenig10) [inline app/models/event.rb]: Change this scope to "displayable".

File: app/views/events/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<% if can?(:create, @event) %>
```
Comment (pkoenig10) [inline app/views/events/edit.html.erb]: Don't do this, use CanCan (see Chase's comment).

File: app/views/events/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<p id="notice"><%= notice %></p>
+
+<% if can?(:read, @event) %>
```
Comment (pkoenig10) [inline app/views/events/index.html.erb]: Don't do this, use CanCan (see Chase's comment).

File: app/views/events/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<% if can?(:create, @event) %>
```
Comment (pkoenig10) [inline app/views/events/new.html.erb]: Don't do this, use CanCan (see Chase's comment).

File: app/models/ability.rb [hunk 6]
```diff
@@ -64,7 +64,8 @@ def initialize(user)
     if user.participant.is_scc?
       can :read, :all
       cannot :read, Role
-
+      can [:create, :update], Event
+      can [:create, :update], EventType
```
Comment (pkoenig10) [inline app/models/ability.rb]: Keep these alphabetized and don't remove the empty line.

File: app/views/events/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+</dl>
+
+
+<%= link_to t('.back', :default => t("helpers.links.back")),
+          events_path, :class => 'btn btn-default' %>
+
+<% if can?(:update, @task) %>
+  <%= link_to t('.edit', :default => t("helpers.links.edit")),
+          edit_event_path(@event), :class => 'btn btn-primary' %>
+<% end %>
```
Comment (pkoenig10) [inline app/views/events/show.html.erb]: Completely redo this view to match the show views found else where in Binder.  For example, mimic the tasks show view.

**Comment by pkoenig10** [comment]
@RONNCC  see my two minor comments.  Also three major things.

1) You did not create a database migration for your new model.  So this model is not created in the database (not sure how that table was in your dev environment unless you manually added it...)
2) Add "load_and_authorize_resource" to your events controller so it uses Cancan.  Then remove the lines in the controller where you fetch the event (Chase's comment) and remove the if-else authorization checks from the views.
3) Your views are basically the default ones generated by Rails.  While these look fine, the views need to be consistent with the rest of the app and use Bootstrap.  Look at any other views (tasks or charges are probably most similar here) and mimic those for events.

**Comment by RONNCC** [comment]
@pkoenig10  rather than a migration, I wrote it directly into the db schema so it initializes at first creation rather than cluttering up the migrations - you should be able to see it if you do rake db:reset (throws away db, creates new one based on schema, and then seeds that) - that way you can add seeds for the EventTypes in seeds.rb

**Comment by purelogiq** [comment]
@RONNCC You'll need to make a migration. The schema file isn't meant to be changed (as you will see at the [top of the file in the comments](https://github.com/RONNCC/binder-app/blob/event_log_2/db/schema.rb#L2)). For example if in the future I make a new migration and do rake db:migrate, the schema file will automatically be updated and will no longer have your change (because it is auto generated by rails based on the migration files). rake db:reset isn't meant to be used outside of a dev environment (or other special cases). Definitely need the migrations. You can do `rails generate migration CreateEventLog` to create the appropriately named migration file and add to it manually if you wish. See [http://guides.rubyonrails.org/active_record_migrations.html](http://guides.rubyonrails.org/active_record_migrations.html)

**Comment by RONNCC** [comment]
@purelogiq  updated

File: app/views/events/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
```
Comment (minigeek) [inline app/views/events/index.html.erb]: This should be in the controller

File: app/views/events/edit.html.erb (hunk 3 above)
Comment (minigeek) [inline app/views/events/edit.html.erb]: This should be in the controller

File: app/views/events/new.html.erb (hunk 5 above)
Comment (minigeek) [inline app/views/events/new.html.erb]: This should be in the controller

File: app/views/layouts/_sidebar.html.erb [hunk 9]
```diff
@@ -211,3 +211,33 @@
   <% end %>
 <% end %>
 
+<%# Change to true to enable queues %>
```
Comment (minigeek) [inline app/views/layouts/_sidebar.html.erb]: Should this be SCC only? If so it should be inside of the `if` block above (which ends at like 212)

**Pull request merged**